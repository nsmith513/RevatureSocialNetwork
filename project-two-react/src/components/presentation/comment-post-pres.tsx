import React, { useEffect, useState } from 'react';
import axios  from 'axios';
import '../styles/comment-post-pres.css';

interface IProps {
    
}

export const CommentPost: React.FC<IProps> = (props:IProps) => {

    const [comment, setComment] = useState('');

    const onContentChange = (evt:any) => {
        setComment(evt.target.value);
    }

    const handleSubmit = (evt:any) => {
        evt.preventDefault();

        const commentObj = {
            content: comment,
            user: 1,
            post: 2
        
        }
        axios.post(process.env.REACT_APP_API_URL + 'ProjectTwoSTS/api/post/comment',commentObj)
        .then(res => console.log(res))
        .catch(err => console.log(err)
        );
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="d-flex justify-content-center">
            <div className="card register-card">
                <div className="card-header">
                    <h3>Comment</h3>
                </div>
                <div className="card-body">
                    <div className="mb-3">
                        <textarea  onChange={onContentChange} value={comment} rows={5} className="form-control" name="description" placeholder="Write a post..."></textarea>
                    </div>
                    
                    <div className="form-group">
                        
                    </div>

                    <button type="submit" className="btn btn-primary" id="input-creds">Submit</button>
                    </div>
                </div>
            </div>
        </form>
    );
}