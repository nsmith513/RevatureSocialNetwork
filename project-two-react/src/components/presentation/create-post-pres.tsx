import React, { FormEvent, RefObject, useEffect, useState } from 'react';
import axios  from 'axios';
import '../styles/create-post-pres.css';

interface IProps {
    fileInput: any;
    handleSubmit: ((event: FormEvent<HTMLFormElement>) => void) | undefined;
    setPostValue(value: string): void;
    postContent: string;

}

export const CreatePost: React.FC<IProps> = (props:IProps) => {



    return (
        <form onSubmit={props.handleSubmit}>
            <div className="container">
                <div className="card create-post">

                    <div id="inputsContainer1" className="col-md-8 col-md-offset-2">

                        <h1 className="createPostHeader">Create a post</h1>

                        <div id="create-post-card">
                            <div className="form-group">
                                <textarea value={props.postContent} onChange={(evt) => props.setPostValue(evt.target.value)} rows={5} className="form-control" name="description" placeholder="Write a post..."></textarea>
                            </div>

                            <div>
                            <label htmlFor="formFileSm" className="form-label">Add an image</label>
                            <input ref={props.fileInput} className="form-control postInputForm form-control-sm" id="formFileSm" type="file" />
                            </div>
                            
                            <br/>

                            <div className="form-group">
                            <button className="btn btn-outline-danger cancleCreateButton">
                                    Cancel
                                </button>
                                <button type="submit" className="btn btn-outline-success createPostButton">
                                    Create
                                </button>

                            </div>

                        </div>
                    </div>

                </div>
            </div>
        </form>
    );
}