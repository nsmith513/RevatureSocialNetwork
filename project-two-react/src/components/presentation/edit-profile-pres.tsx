import React from 'react';
import IUser from '../../models/IUser';
import '../styles/edit-profile-pres.css';

interface IUserModel {
        id:number,
        uname:string,
        pwd:string,
        email:string,
        pfp:string
}

interface IProps {
    handleSubmit(event: React.FormEvent<HTMLFormElement>): void;
    setUsername: Function;
    setPassword: Function;
    setEmail: Function;
    username: string;
    password: string;
    email: string,
    userData: IUserModel;
    fileInput:any;
}

export const EditProfilePres: React.FC<IProps> = (props:IProps) => {
    return(
        <>
         <form onSubmit={props.handleSubmit}>
            <div className="card edit-profile-card">
                    <div className="card-body userProfile">

                    <h1>Update Profile</h1>
                        
                            <img className='avatar updateProfileAvatarImage  edit-custom-img' src={props.userData.pfp}/> 

                            <div className="">
                                <label htmlFor="formFileSm" className="form-label">Change your profile picture</label>
                                <input ref={props.fileInput} className="updateProfileImage form-control form-control-sm" id="formFileSm" type="file" />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput1" className="form-label">Change Username</label>
                                <input value={props.username} onChange={(evt) => props.setUsername(evt.target.value)} type="text"  name="username" className="form-control" id="exampleFormControlInput1" placeholder={props.userData.uname} />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput2" className="form-label">Change Email</label>
                                <input value={props.email} onChange={(evt) => props.setEmail(evt.target.value)} type="email"  name="email" className="form-control" id="exampleFormControlInput2" placeholder={props.userData.email} />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput2" className="form-label">Change Password</label>
                                <input value={props.password} onChange={(evt) => props.setPassword(evt.target.value)} type="password"  name="password" className="form-control" id="exampleFormControlInput3" placeholder='password' />
                            </div>

                            <div className="mb-3 profileSubmitButtonContainer">
                                <button type="submit" className="btn btn-outline-light mb-3 updateProfileSubmitButton">Update Info</button>
                            </div>
                                    
                            <br/>
                        
                    </div>
                </div>
            </form>
        </>
    )
}