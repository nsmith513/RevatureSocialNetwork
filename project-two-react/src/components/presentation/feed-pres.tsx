import React from 'react';

export const FeedPres: React.FC = (props) =>{
    return <>{props.children}</>;
}
