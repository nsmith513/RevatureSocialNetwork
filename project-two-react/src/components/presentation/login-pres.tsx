import React, { useState } from 'react';
import '../styles/login-pres.css';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';

interface IProps {
    handleSubmit(event: React.FormEvent<HTMLFormElement>): void;
    setUsername: Function;
    setPassword: Function;
    username: string;
    password: string;
}

export const LoginPres: React.FC<IProps> = (props:IProps) =>{

    return (
        <div className="login">
            <div className="d-flex justify-content-center">
                <div className="card login-card">
                    <div className="card-header">
                        <h3 className="loginCardHeader">User Login</h3>
                    </div>
                    <div className="card-body">

                        <form onSubmit={props.handleSubmit}>
                            <div className="mb-3">

                                <label htmlFor="inputUsername1" className="form-label ">Username</label>
                                <input placeholder="Username" onChange={(evt) => props.setUsername(evt.target.value)} type="text" className="form-control form-control-lg" id="input-username" name="username" value={props.username} />
                            </div>
                            <div className="mb-3">
                                <label htmlFor="inputPassword1" className="form-label">Password</label>
                                <input placeholder="Password" onChange={(evt) => props.setPassword(evt.target.value)} type="password" className="form-control form-control-lg" id="input-password" name="password" value={props.password} />
                            </div>
                            <div className="row">
                                <div className="col">

                                    <button type="submit" className="btn btn-lg btn-outline-light loginSubmitButton" id="input-creds">Submit</button>

                                </div>
                                <div className="col text-right">
                                    <Link to="/register"><button className="btn btn-lg btn-outline-warning registerButton goToRegisterButton">Register</button></Link>
                                </div>
                            </div>
                        </form>

                    </div>
                </div>
            </div>
        </div>
    );

    

}
