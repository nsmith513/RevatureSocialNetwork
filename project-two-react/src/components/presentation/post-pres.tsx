import React from 'react';
import IPost from '../../models/IPost';
import '../styles/post-pres.css';

interface IProps {
    likeButtonClick(event: any): void;
    post: IPost;
    onOff: Boolean;
    likeCount: number;
}

interface IImages{
    id: number,
    link: string
}

export const PostPres: React.FC<IProps> = (props: IProps) => {
    return props.post.user ?
        <>
            <div className="card post-custom-card ">
                <div className="card-body">
                    <div className="clearfix">
                        <img className='avatar pull-left custom-img' src={props.post.user.pfp} />
                        <h5 className="card-title">{props.post.user.uname}</h5>
                    </div>
                    <p className="card-text">{props.post.content}</p>

                    <div>{props.post.images.map((image) => <img className="img-fluid img-thumbnail" key={image.id} src={image.link}/>)}</div>
                    
                    <hr />
                    <button onClick={props.likeButtonClick}
                        style={props.onOff ? { borderColor: 'blue' } : { borderColor: 'grey' }}
                        className="btn btn-link">
                        Like {props.likeCount > 0 ? props.likeCount : ""}
                    </button>
                    {/* <a href="#" className="card-link">Comment</a> */}
                </div>
            </div>
        </> :
        <>
            <h2>No posts</h2>
        </>
}
