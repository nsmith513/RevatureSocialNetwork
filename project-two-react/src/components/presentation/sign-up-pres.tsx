import React, {useState, useRef} from 'react';
import axios from 'axios';
import "../styles/sign-up-pres.css";
const S3 = require('react-aws-s3');

interface IProps {
    handleSubmit(event: React.FormEvent<HTMLFormElement>): void;
    onContentChange(event: any): void;
    setUsername: Function;
    setPassword: Function;
    setPassword2: Function;
    setEmail: Function;
    setPasswordCheck: Function;
    username: string;
    password: string;
    password2:string;
    email: string;
    passwordCheck: string;
    fileInput:any;
}

const config = {
    bucketName: process.env.REACT_APP_BUCKET_NAME,
    region: process.env.REACT_APP_REGION,
    accessKeyId: process.env.REACT_APP_ACCESS_ID,
    secretAccessKey: process.env.REACT_APP_ACCESS_KEY,
};


export const SignUpPres: React.FC<IProps> = (props:IProps) => {
    return (
        <form onSubmit={props.handleSubmit} >
        <div className="d-flex justify-content-center">
            <div className="card register-card registerCard">
                <div className="card-header">
                    <h3 className="registerNewUserHeader">Register</h3>
                </div>
                    <div className="mb-3">
                        <label htmlFor="inputUsername1" className="form-label  ">Username</label>
                        <input placeholder="Username" value={props.username} onChange={(evt) => props.setUsername(evt.target.value)} type="text" className="form-control form-control-lg" id="input-username" name="username" required/>
                    </div>
                    <div className="mb-3">
                        <label htmlFor="inputEmail1" className="form-label">Email</label>
                        <input placeholder="Email" value={props.email} onChange={(evt) => props.setEmail(evt.target.value)} type="email" className="form-control form-control-lg" id="input-email" name="email" required/>
                    </div>
                    <div className="mb-3" style={{color:"red"}}>
                        {props.passwordCheck}
                    </div>
                    <div className="mb-3">
                        <label htmlFor="inputPassword1" className="form-label">Password</label>
                        <input placeholder="Password" value={props.password} onChange={(evt) => props.setPassword(evt.target.value)} type="password" className="form-control form-control-lg" id="input-password" name="password" required/>
                    </div>
                    <div className="mb-3">
                        <label htmlFor="inputConfirmPassword1" className="form-label">Confirm Password</label>
                        <input placeholder="Confirm Password" value={props.password2} onChange={(evt) => props.setPassword2(evt.target.value)}type="password" className="form-control form-control-lg" id="input-confirm-password" name="password2" required/>
                    </div>

                    <br/>
                    <div className="form-group">
                        <label htmlFor="profilePicture">Add a Profile Picture</label>
                        <input ref={props.fileInput} type="file" className="form-control-file" id="exampleFormControlFile1" required/>
                    </div>
                    <br/>


                    <button type="submit" className="btn btn-lg btn-outline-warning" id="input-creds">Create Account</button>
                    </div>
                </div>
            </form>
    );
}