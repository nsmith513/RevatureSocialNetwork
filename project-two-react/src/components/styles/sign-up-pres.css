.card.register-card {
  margin: 2vh 1rem 0 2rem;
  width: 30rem;
  background-color: var(--col-e);
}

.registerNewUserHeader {
  font-size: xxx-large;
  font-weight: bold;
  -webkit-text-stroke-width: 0.1px;
  -webkit-text-stroke-color: #477387;
}

.registerCard {
  padding: 15px;
  margin-bottom: 20px;
}
