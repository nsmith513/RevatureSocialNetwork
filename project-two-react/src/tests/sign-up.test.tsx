import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {SignUpPres} from '../components/presentation/sign-up-pres';

Enzyme.configure({ adapter: new Adapter() })


const fileInput = null;
const handleSubmit = (evt: any) => {}
const onContentChange = (event: any) => {}
const setUsername = () => {}
const setPassword = () => {}
const setPassword2 = () => {}
const setEmail = () => {}
const setPasswordCheck = () => {}
const username = ''
const password = ''
const password2 =''
const email =''
const passwordCheck = ''

describe('<SignUpPres/>', () => {
    const container = shallow(<SignUpPres 
        handleSubmit={handleSubmit}
        onContentChange={onContentChange}
        setUsername={setUsername}
        setPassword={setPassword}
        setPassword2={setPassword2}
        setEmail={setEmail}
        setPasswordCheck={setPasswordCheck}
        username={username}
        password={password}
        password2={password2}
        email={email}
        passwordCheck={passwordCheck}
        fileInput={fileInput}
    />);

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});