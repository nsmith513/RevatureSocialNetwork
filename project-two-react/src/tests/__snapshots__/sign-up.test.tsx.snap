// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPres/> should match the snapshot 1`] = `"<form><div class=\\"d-flex justify-content-center\\"><div class=\\"card register-card registerCard\\"><div class=\\"card-header\\"><h3 class=\\"registerNewUserHeader\\">Register</h3></div><div class=\\"mb-3\\"><label for=\\"inputUsername1\\" class=\\"form-label  \\">Username</label><input type=\\"text\\" placeholder=\\"Username\\" value=\\"\\" class=\\"form-control form-control-lg\\" id=\\"input-username\\" name=\\"username\\" required=\\"\\"/></div><div class=\\"mb-3\\"><label for=\\"inputEmail1\\" class=\\"form-label\\">Email</label><input type=\\"email\\" placeholder=\\"Email\\" value=\\"\\" class=\\"form-control form-control-lg\\" id=\\"input-email\\" name=\\"email\\" required=\\"\\"/></div><div class=\\"mb-3\\" style=\\"color:red\\"></div><div class=\\"mb-3\\"><label for=\\"inputPassword1\\" class=\\"form-label\\">Password</label><input type=\\"password\\" placeholder=\\"Password\\" value=\\"\\" class=\\"form-control form-control-lg\\" id=\\"input-password\\" name=\\"password\\" required=\\"\\"/></div><div class=\\"mb-3\\"><label for=\\"inputConfirmPassword1\\" class=\\"form-label\\">Confirm Password</label><input type=\\"password\\" placeholder=\\"Confirm Password\\" value=\\"\\" class=\\"form-control form-control-lg\\" id=\\"input-confirm-password\\" name=\\"password2\\" required=\\"\\"/></div><br/><div class=\\"form-group\\"><label for=\\"profilePicture\\">Add a Profile Picture</label><input type=\\"file\\" class=\\"form-control-file\\" id=\\"exampleFormControlFile1\\" required=\\"\\"/></div><br/><button type=\\"submit\\" class=\\"btn btn-lg btn-outline-warning\\" id=\\"input-creds\\">Create Account</button></div></div></form>"`;
